lambda/service/store_dynamodb/store_dynamodb.go

```diff
@@ -44,12 +48,34 @@ func (r *AccountDatabaseStore) GetById(ctx context.Context, uuid string) (Accoun
 	})
 	if err != nil {
 		log.Printf("couldn't get info about %v. Here's why: %v\n", uuid, err)
-	} else {
-		err = attributevalue.UnmarshalMap(response.Item, &account)
-		if err != nil {
-			log.Printf("couldn't unmarshal response. Here's why: %v\n", err)
-		}
+		return account, err
+	}
+
+	err = attributevalue.UnmarshalMap(response.Item, &account)
+	if err != nil {
+		log.Printf("couldn't unmarshal response. Here's why: %v\n", err)
+		return account, err
+	}
+
+	return account, nil
+}
+
+func (r *AccountDatabaseStore) Get(ctx context.Context) ([]Account, error) {
+	accounts := []Account{}
+
+	response, err := r.DB.Scan(ctx, &dynamodb.ScanInput{
+		TableName: aws.String(r.TableName),
+	})
+	if err != nil {
+		log.Printf("couldn't get accounts. Here's why: %v\n", err)
```

Comment:
This error is logged here and then again by the caller in `GetAccountsHandler`. Similarly with other errors in the DynamoDB store. It was drilled into me at some point to only log errors once, at the point of handling. To avoid repetition in the logs. I think it's enough to log this in the handler when it is called and not here.

Is the idea of the logging here, to help distinguish between an error getting the accounts vs an unmarshal error? An alternative is to wrap the error instead of logging it:
`return accounts, fmt.Errorf("error getting accounts: %w", err)`

Reply:
wrapped errors, thanks @jfromme